zk/stages/stage_sequence_execute.go

```diff
@@ -47,6 +44,83 @@ func SpawnSequencingStage(
 		defer tx.Rollback()
 	}
 
+	lastBatch, err := stages.GetStageProgress(tx, stages.HighestSeenBatchNumber)
+	if err != nil {
+		return err
+	}
+
+	highestBatchInDS, err := cfg.datastreamServer.GetHighestBatchNumber()
+	if err != nil {
+		return err
+	}
+
+	if lastBatch >= highestBatchInDS {
+		err = sequencingStageStep(s, u, tx, ctx, cfg, quiet, nil)
+		if err != nil {
+			return err
+		}
+	} else {
+		log.Info(fmt.Sprintf("[%s] Last batch %d is lower than highest batch in datastream %d, resequencing...", s.LogPrefix(), lastBatch, highestBatchInDS))
+
+		latestForkId, err := stages.GetStageProgress(tx, stages.ForkId)
+		if err != nil {
+			return err
+		}
+		dsClient := client.NewClient(ctx, fmt.Sprintf("%s:%d", cfg.zk.DataStreamHost, cfg.zk.DataStreamPort), cfg.zk.DatastreamVersion, cfg.zk.L2DataStreamerTimeout, uint16(latestForkId))
+
+		batches, err := dsClient.ReadBatches(lastBatch+1, highestBatchInDS)
```

Comment:
Correct me if I'm wrong, but if we have reached this point, it means that `highestBatchInDS < lastBatch` (negation of this condition https://github.com/cffls/erigon/blob/5f9d291b33150793a9af43330c6caf59454430a3/zk/stages/stage_sequence_execute.go#L57), so it doesn't seem like the interval for fetching batches is correct.

Reply:
I think when we reach this point, `lastBatch` will be smaller than `highestBatchInDS`, but will double check.

Reply:
Yes, you are right, my bad!